packages/fx-core/src/component/driver/deploy/azure/AzureZipDeployDriver.ts

```diff
@@ -0,0 +1,212 @@
+import { TokenCredential } from "@azure/core-auth";
+import { IProgressHandler, LogProvider, UserInteraction } from "@microsoft/teamsfx-api";
+import { getLocalizedMessage, ProgressMessages } from "../../../messages";
+import { DeployConstant } from "../../../constant/deployConstant";
+import { createHash } from "crypto";
+import { default as axios } from "axios";
+import { DeployExternalApiCallError } from "../../../error/deployError";
+import { HttpStatusCode } from "../../../constant/commonConstant";
+
+export class AzureZipDeployDriver extends AzureDeployDriver {
+  pattern =
+    /\/subscriptions\/([^\/]*)\/resourceGroups\/([^\/]*)\/providers\/Microsoft.Web\/sites\/([^\/]*)/i;
+  private readonly serviceName: string;
+  protected helpLink = "https://aka.ms/teamsfx-actions/azure-app-service-deploy";
+  protected summaries: string[];
+  protected summaryPrepare: string[];
+  protected zipBuffer: Buffer | undefined;
+
+  constructor(
+    args: unknown,
+    context: DriverContext,
+    serviceName: string,
+    summaries: string[],
+    summaryPrepare: string[]
+  ) {
+    super(args, context);
+    this.serviceName = serviceName;
+    this.summaries = summaries;
+    this.summaryPrepare = summaryPrepare;
+  }
+
+  async azureDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredential
+  ): Promise<void> {
+    await this.progressBar?.start();
+    const cost = await this.zipDeploy(args, azureResource, azureCredential);
+    await this.progressBar?.next(ProgressMessages.restartAzureService);
+    await this.restartFunctionApp(azureResource);
+    await this.progressBar?.end(true);
+    if (cost > DeployConstant.DEPLOY_OVER_TIME) {
+      await this.context.logProvider?.info(
+        getLocalizedMessage(
+          "driver.deploy.notice.deployAcceleration",
+          "https://aka.ms/teamsfx-config-run-from-package"
+        ).localized
+      );
+    }
+  }
+
+  protected prepare: (args: DeployStepArgs) => Promise<void> = async (args: DeployStepArgs) => {
+    await this.progressBar?.next(ProgressMessages.packingCode);
+    await this.packageToZip(args, this.context);
+  };
+
+  /**
+   * deploy to azure app service or azure function use zip deploy method
+   * @param args local file needed to be deployed
+   * @param azureResource azure resource info
+   * @param azureCredential azure user login credential
+   * @return the zip deploy time cost
+   * @protected
+   */
+  public async zipDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredential
+  ): Promise<number> {
+    await this.progressBar?.next(ProgressMessages.packingCode);
+    const zipBuffer = await this.packageToZip(args, this.context);
+    await this.progressBar?.next(ProgressMessages.getAzureAccountInfoForDeploy);
+    await this.context.logProvider.debug("Start to get Azure account info for deploy");
```

Comment:
Nit: How about extracting those debug messages out and grouping them into a constant file?